### PR TITLE
Upgrading to vendure v2

### PR DESCRIPTION
**Pull request by DanielBiegler**
This is my PR for holding all the needed changes to make Vendure v2 work with this storefront.

Currently using a brand new Vendure `v2.0.0-beta.2` instance for testing.

Relevant Links:

- [Vendure Repo](https://github.com/vendure-ecommerce/vendure/discussions/1991)
- [Qwik Storefront](https://github.com/vendure-ecommerce/storefront-qwik-starter/issues/99)

As a helpful overview, I'll list the mentioned differences from [Vendures Issue](https://github.com/vendure-ecommerce/vendure/discussions/1991):

## Checklist

- [ ] Vendure TypeScript API Changes
  - [x] Custom Order / Fulfillment / Payment processes
    - Irrelevant for storefront
  - [x] OrderItem no longer exists: data of OrderItem is now transferred to OrderLine
    - [x] `OrderHistoryItem` FIXED!
    - [x] "the following APIs which previously used OrderItem arguments have now changed `FulfillmentHandler`, `ChangedPriceHandlingStrategy`, `PromotionItemAction`, `TaxLineCalculationStrategy`"
      - Irrelevant / We dont use these
  - [x] ProductVariant stock changes
    - "`ProductVariant.stockOnHand` and `ProductVariant.stockAllocated` properties no longer exist on the ProductVariant entity" - We dont use these. We only use `stockLevel` for showing our `StockLevelLabel` component on product pages.
  - [x] Other breaking API changes
    - [x] All monetary values are now represented in the GraphQL APIs with a new Money scalar type. - FIXED!
  - [ ] TODO FURTHER CHANGES?

fixes https://github.com/vendure-ecommerce/storefront-remix-starter/issues/40

### CODE REVIEW

**Comment by DanielBiegler** [comment]
Fixed the orderhistory. It now uses the new FulfillmentLines. Also added placeholder package-tracking buttons.

@michaelbromley is there a readonly demo URL for Vendure v2 maybe? Then we could fix the netlify deployments for public testing. :-)

| Header | Header |
|--------|--------|
| image | image |

File: app/providers/orders/order.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ gql`
   }
 `;
 gql`
-  mutation setOrderShippingMethod($shippingMethodId: ID!) {
+  mutation setOrderShippingMethod($shippingMethodId: [ID!]!) {
```
Comment (DanielBiegler) [inline app/providers/orders/order.ts:88]: Careful here, I only "fixed" the type so we can compile but havent looked into the usage yet.

**Comment by michaelbromley** [comment]
@DanielBiegler great that you are working on this! Currently I don't have a v2 demo running anywhere. I might set one up after the next beta release once I've done the bulk of the Admin UI work I'm in the middle of.

File: app/providers/orders/order.ts (hunk 1 above)
Comment (DanielBiegler) [reply app/providers/orders/order.ts:88]: Mutation appearently accepts `string | string[]` so we're good here

**Comment by dylviz** [comment]
Now I'm seeing an error on the /checkout route. Seems like an error on my backend. I noticed a new commit on the major branch 3 days ago maybe it's because I don't have this update

**Comment by kyunal** [comment]
I finally gave it a test run and looked over the changes, everything looks good! I would merge this into a v2 branch for now and would place a hint in the readme file. We can merge everything into main once V2 is released and a demo instance is available.

@DanielBiegler Just mark the PR as ready for review if this sounds good to you and I'll go ahead and do the rest. Thanks for the contribution :raised_hands:

**Comment by zolzaya** [comment]
I tested it with my current project it's working good :)

**Comment by michaelbromley** [comment]
👏 Thank you everyone involved!

**Pull request merged**